sgl-kernel/csrc/gemm/silu_and_mul_per_token_group_quant_fp8.cu

```diff
@@ -0,0 +1,189 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/util/Float8_e4m3fn.h>
+
+#include <cmath>
+#include <flashinfer/vec_dtypes.cuh>
+#include <iostream>
+
+#include "utils.h"
+
+__device__ __forceinline__ float silu(const float& val) {
+  return val / (1.0f + __expf(-val));
+}
+
+template <typename T, typename DST_DTYPE, bool IS_COLUMN_MAJOR = false>
+__global__ void silu_and_mul_per_token_group_quant_8bit_kernel(
+    T* __restrict__ input,
+    const int d,
+    void* __restrict__ output_q,
+    float* __restrict__ output_s,
+    const int group_size,
+    const int num_groups,
+    const int groups_per_block,
+    const int groups_per_token,
+    const float eps,
+    const float min_8bit,
+    const float max_8bit,
+    const int scale_num_rows = 0,
+    const int scale_stride = 0) {
+  const int threads_per_group = 16;
+  const int local_group_id = threadIdx.x / threads_per_group;
+  const int lane_id = threadIdx.x % threads_per_group;
+
+  const int block_group_id = blockIdx.x * groups_per_block;
+  const int global_group_id = block_group_id + local_group_id;
+  const int token_id = global_group_id / groups_per_token;
+  const int group_in_token_offset = global_group_id % groups_per_token;
+  const int block_group_offset = (token_id * groups_per_token * 2 + group_in_token_offset) * group_size;
+  float local_absmax = eps;
+  T* group_input = input + block_group_offset;
+  DST_DTYPE* group_output = static_cast<DST_DTYPE*>(output_q) + global_group_id * group_size;
+  float* scale_output;
+
+  if constexpr (IS_COLUMN_MAJOR) {
+    const int row_idx = global_group_id / scale_num_rows;
+    const int col_idx = global_group_id % scale_num_rows;
+    scale_output = output_s + (col_idx * scale_stride + row_idx);
+  } else {
+    scale_output = output_s + global_group_id;
+  }
+
+  constexpr uint32_t vec_size = 16 / sizeof(T);
+  using vec_t = flashinfer::vec_t<T, vec_size>;
+
+  const int32_t num_vec_elems = group_size / vec_size;
+
+  for (int32_t i = lane_id; i < num_vec_elems; i += 16) {
+    vec_t x_vec, y_dec;
+    x_vec.cast_load(group_input + i * vec_size);
+    y_dec.cast_load(group_input + i * vec_size + d);
+#pragma unroll
+    for (uint32_t j = 0; j < vec_size; ++j) {
+      x_vec[j] = silu(static_cast<float>(x_vec[j])) * static_cast<float>(y_dec[j]);
+      float abs_val = fabsf(x_vec[j]);
+      local_absmax = fmaxf(local_absmax, abs_val);
+    }
+    x_vec.cast_store(group_input + i * vec_size);
```

Comment:
Casting x_vec which datatype is fp16/bf16 to 8bit may degrade precision, especially for int8.
IDK why the unittest is passed but we should consider to save temp result in shared_mem.

Reply:
Very good advice. it does make for a increase in readability. I've updated it. Thank you！